payjoin/src/sender/mod.rs

```diff
@@ -38,14 +38,14 @@ type InternalResult<T> = Result<T, InternalValidationError>;
 /// Builder for sender-side payjoin parameters
 ///
 /// These parameters define how client wants to handle PayJoin.
-pub struct Params {
+pub struct Configuration {
     disable_output_substitution: bool,
     fee_contribution: Option<(bitcoin::Amount, Option<usize>)>,
```

Comment:
Is there any value putting params here as a field?

Reply:
If this configuration is going to hold this data then yes, I think it makes sense to be part of this. The idea of a builder to expose this interface in an accessible way rather than expose `optional_parameters::Params` as pub makes sense to me.

I'm starting to think of this more as a `sender::RequestBuilder` that can turn into a `(Request, Context)` when you call `from_psbt_and_uri`. Maybe the signature for that method should be `request_builder.create_pj_request(psbt, pj_url) -> Result(<(Request, Context)>` instead of `pj_url.create_pj_request(psbt, pj_params: Configuration) -> Result<(Request, Context)>.

Then create_pj_request could validate the endpoint `check_pj_supported()` during that call, too

Reply:
The builder could have `.psbt()`, `pj_uri()`, `.build()` methods in addition to the existing one's like `.non_incentivizing()`  to more closely resemble a builder pattern, too

Reply:
Yeah, builder pattern kinda makes sense. We might need more than one to implement type state, since both URI and PSBT are mandatory. Alternatively we could just accept both in a single method.